### PR TITLE
add yesno_box_no and yesno_box_yes

### PR DESCRIPTION
**Pull request by szaimen**
POC, fix part of #1405
The function works as expected (was already tested) but the cases where it is needed are still missing (I don't know where it was no by default before)

### CODE REVIEW

**Comment by enoch85** [comment]
You can check the defaults here: https://github.com/nextcloud/vm/pull/1401/files

**Comment by szaimen** [comment]
> You can check the defaults here: https://github.com/nextcloud/vm/pull/1401/files

I've gone through the list and added the new function to all places, where I would say, that it makes sense.

**Comment by szaimen** [comment]
Please wait before merging

**Comment by szaimen** [comment]
I think it is ready now.

File: apps/onlyoffice_docker.sh [hunk 1]
```diff
@@ -146,7 +146,7 @@ fi
 if [ "$(apache2ctl -M | grep evasive)" != "" ]
 then
     msg_box "We noticed that 'mod_evasive' is installed which is the DDOS protection for webservices. It has comptibility issues with OnlyOffice and you can now choose to disable it."
-    if ! yesno_box "Do you want to disable DDOS protection?"
+    if ! yesno_box_defaultno "Do you want to disable DDOS protection?"
```
Comment (enoch85) [inline apps/onlyoffice_docker.sh:149]: I think default should be yes here.

File: apps/onlyoffice_integrated.sh [hunk 2]
```diff
@@ -172,7 +172,7 @@ fi
 if [ "$(apache2ctl -M | grep evasive)" != "" ]
 then
     msg_box "We noticed that 'mod_evasive' is installed which is the DDOS protection for webservices. It has comptibility issues with OnlyOffice and you can now choose to disable it."
-    if ! yesno_box "Do you want to disable DDOS protection?"
+    if ! yesno_box_defaultno "Do you want to disable DDOS protection?"
```
Comment (enoch85) [inline apps/onlyoffice_integrated.sh:175]: Yes here as well

**Comment by enoch85** [comment]
```
yesno_box() {
    [ -n "$2" ] && local SUBTITLE=" - $2"

    # Check if yes or no
    if [ "$3" == "yes" ]
    then
        YESNO="--yesno"
    elif [ "$3" == "no" ]
    then
        YESNO="--defaultno --yesno"
    else
        YESNO="--yesno"
    fi

    # Final command
    if (whiptail --title "$TITLE$SUBTITLE" $YESNO "$1" "$WT_HEIGHT" "$WT_WIDTH")
    then
        return 0
    else
        return 1
    fi
}
```

I still don't get where `$2` comes from?

Travis will probably fail on the above since `$YESNO` can't be with `" "`, but we can work around it. It would be more neat to have everything in one function.

**Comment by szaimen** [comment]
> I still don't get where `$2` comes from?

It is already used inside the smb-mount script and e.g. here:
https://github.com/nextcloud/vm/blob/2cc0f8a17a656ac3b2713d336fbaefd3a88d8609/apps/smbmount.sh#L87
The idea was to be able to add a custom Subtitle per msg_box, input_box or yesno_box in order create the possibility to customize the title further.

> Travis will probably fail on the above since `$YESNO` can't be with `" "`, but we can work around it. It would be more neat to have everything in one function.

This is what I meant when I thought about if this will work.

And BTW: I still think, that this:
`if yesno_box_defaultno "That is the message"`
looks much better than this
`if yesno_box "That is the message" "" "no"`

Or a bit differently: Why don't you want to add another short function and instead make the existing one more complicated?

File: apps/onlyoffice_docker.sh (hunk 1 above)
Comment (szaimen) [reply apps/onlyoffice_docker.sh]: done

File: apps/onlyoffice_integrated.sh (hunk 2 above)
Comment (szaimen) [reply apps/onlyoffice_integrated.sh]: done

**Comment by szaimen** [comment]
A different idea: what about changing the name of` yesno_box_defaultno` to `yesno_box_no` ? 
In order to be consistent, we could rename `yesno_box` then to `yesno_box_yes`
What do you think about that?

**Comment by enoch85** [comment]
> What do you think about that?

That's better, but can't be merged until next release as the `static_ip` use the old function and the `lib` file is also containing the old function.

But yes, I think that's better.

So, let's change `defaultno` to `no` for now, and change the name of the regular `yesno_box_yes` later.

**Comment by szaimen** [comment]
I will work on this tomorrow. Don't have time today

**Comment by szaimen** [comment]
I think this is ready now.
I have renamed yesno_box_defaultno to yesno_box_no.
Additionally, I have created yesno_box_yes and renamed all appearances of yesno_box to this.
But there should be no problem since I have not deleted the current function yesno_box and we can just delete it with NC20 or so.

**Comment by szaimen** [comment]
Is there anything missing here or is there still something to discuss?

**Comment by enoch85** [comment]
> Is there anything missing here or is there still something to discuss?

Nope, I think it looks good. :+1:

**Comment by enoch85** [comment]
Tested the current VM (19.0.2) without internet, and all is good. It doesn't bug out after this merge.

**Pull request merged**